pkg/centralconfig/central_config_cache_init.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package centralconfig
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/discovery"
+	"github.com/vmware-tanzu/tanzu-cli/pkg/globalinit"
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+)
+
+// The Central Configuration feature uses a central_config.yaml file that gets downloaded
+// along with the plugin inventory cache and is stored in the cache.  Older versions of the
+// CLI (< 1.3.0) do not include this file in the cache.  It is therefore possible that
+// the plugin inventory cache was setup by an older version of the CLI and is missing
+// the central_config.yaml file.  In such a case, the digest of the cache will still indicate
+// that the latest plugin inventory is present and the content of the cache will not be refreshed
+// until the plugin inventory data changes in the central repo itself.  In such a case, to be able
+// to benefit from the central configuration feature once the CLI is upgraded to >= 1.3.0
+// we need to fix the cache.  This initializer checks if the central_config.yaml file is
+// present in the cache and if not, it invalidates the cache.
+
+func init() {
+	globalinit.RegisterInitializer(triggerForInventoryCacheInvalidation, invalidateInventoryCache)
+}
+
+// triggerForInventoryCacheInvalidation returns true if the central_config.yaml file is missing
+// in the plugin inventory cache.
+func triggerForInventoryCacheInvalidation() bool {
+	sources, err := config.GetCLIDiscoverySources()
+	if err != nil {
+		// No discovery source
+		return false
+	}
+
+	for _, source := range sources {
+		centralConfigFile := filepath.Join(common.DefaultCacheDir, common.PluginInventoryDirName, source.OCI.Name, constants.CentralConfigFileName)
+
+		if _, err := os.Stat(centralConfigFile); os.IsNotExist(err) {
+			// As soon as we find a source that doesn't have a central_config.yaml file,
+			// we need to perform some initialization.
+			return true
+		}
+	}
```

Comment:
You are right.  There will only be one source: `default`.
However, if we ever start supporting more than one source, then we'd have to change this logic, so I figured it wasn't too hard to make the code future-proof.

Note that this logic does not include sources set using `TANZU_CLI_ADDITIONAL_PLUGIN_DISCOVERY_IMAGES_TEST_ONLY` since they are not in the config file.  Those sources are not important anyway for the central config since only the central config of the `default` source is used.

Reply:
But now that I write this I realize that the recommended version feature is not future-proof if we have multiple (official) plugin sources (in the future).
Do you think we should have an issue to track that?